### DISCUSSION TITLE
TextField is not visible on android devices

### DISCUSSION THREAD

**Discussion by iAbdulQadirKhan** [axmolengine/axmol#1640] [General]
I have created a text. This text is visible on IOS devices but not on android devices. It is working fine on cocos2d-v3. We are using Cocos Studio.

```
Node* const probabilityCellNode = CSLoader::createNode("TestMenuNumberCell.csb");
ui::TextField* text = probabilityCellNode->getChildByName<ui::TextField*>("txtCell");
text->setVisible(true);
text->setString("Text");
```

**Comment by rh101** [comment]
Which version of Axmol?
Have you tried creating a `ui::TextField` manually to see if it works (one that isn't created in Cocos Studio)?

**Reply by iAbdulQadirKhan** [reply]
Version: 2.0.0
Manually it is visible.

**Reply by rh101** [reply]
This may just be stating the obvious, but have you used a debugger to figure out why it's not visible? Check both the CPP and Java code.

**Reply by iAbdulQadirKhan** [reply]
This issue is related to the font. Now I am creating text field with font. But facing a problem related to text input. This problem is also present in manually created text fields. I'm not able to enter text into it. When I click on the text field the keyboard doesn't appear.

```
    TextField* text = TextField::create("input words here", "impact.ttf", 30);
    text->setTouchEnabled(true);
    text->setEnabled(true);
```

**Reply by rh101** [reply]
Look, just for future reference, when reporting an issue, not providing all the relevant information to diagnose an issue isn't doing you any favors.  

For example:

- Which version of Android? 
- What device? 
- Does it happen in emulators too?
- Does it happen with any font, or just that specific font? If it's just that font, put a link to the font in your post so we can download it.

**Reply by iAbdulQadirKhan** [reply]
[https://github.com/axmolengine/axmol/pull/1429](https://github.com/axmolengine/axmol/pull/1429)
I think this problem is fixed in this pull request. I have'n this changes. Now it is working fine.

**Reply by rh101** [reply]
To make it easier for yourself, always test with the latest version to see if these issues have been fixed.

**Comment by iAbdulQadirKhan** [comment]
@rh101 Thanks for your response.
Recently we updated coco2d-v3 to axmol 2.0.0. we are currently fixing the issues. Will be updated in future.
